app/controllers/concerns/course/activities_concern.rb

```diff
@@ -0,0 +1,16 @@
+module Course::ActivitiesConcern
+  extend ActiveSupport::Concern
+
+  # Load recent activity feeds
+  #
+  # @param [Course] course course which activities are related to
+  # @param [Integer] number number of activities that will be loaded
+  #
+  # @return [Array<Activity>]
+  def recent_activities(course, number: nil)
+    result = Activity.joins(:notification).where('notifications.activity_feed = true').
```

Comment:
Here is a big problem that makes me spend the whole day last Friday. I also discussed with @allenwq , finally we have to write this raw sql. I don't know why. But it is about the usage of `polymorphic ` in gem's table and the necessary `joins` in this query. @lowjoel